app/scripts/modules/amazon/src/function/configure/Network.tsx

```diff
@@ -0,0 +1,248 @@
+        return this.setState({ subnets, securityGroups });
+      });
+  }
+
+  private handleSubnetUpdate = (options: Array<Option<string>>) => {
+    const subnetsSelected = options.map(o => o.value);
+    this.props.formik.setFieldValue('subnetIds', subnetsSelected);
+  };
+
+  private handleSecurityGroupsUpdate = (options: Array<Option<string>>) => {
+    const sgSelected = options.map(o => o.value);
+    this.props.formik.setFieldValue('securityGroupIds', sgSelected);
+  };
+
+  private handleVpcChange = (vpcId: string): void => {
+    this.props.formik.setFieldValue('vpcId', vpcId);
+    const { subnets } = this.state;
+    const subs = subnets.filter(function(s: ISubnetOption) {
+      return s.vpcId.includes(vpcId);
+    });
+    this.setState({ subnets: subs });
+  };
+
+  private toSubnetOption = (value: ISubnetOption): Option<string> => {
+    return { value: value.subnetId, label: value.subnetId };
+  };
+
+  private getSecurityGroupsByVpc = (sgs: ISecurityGroupsByAccountSourceData): Array<Option<string>> => {
+    const { values } = this.props.formik;
+    const sgOptions: Array<Option<string>> = [];
+    /** Get security groups that belong to current selected account */
+    forOwn(sgs, function(sgByAccount, acc) {
+      if (acc === values.credentials) {
+        /** Get security groups that fall under the provider 'aws' */
+        forOwn(sgByAccount, function(sgByRegion, provider) {
+          if (provider === 'aws') {
+            /** Get security groups that are under the current selected region */
+            forOwn(sgByRegion, function(groups, region) {
+              if (region === values.region) {
+                /** Get security groups that fall under the current selected VPC */
+                groups.forEach(function(group) {
+                  if (group.vpcId === values.vpcId) {
+                    sgOptions.push({ value: group.id, label: group.name });
+                  }
+                });
+              }
+            });
+          }
+        });
+      }
+    });
+    return sgOptions;
+  };
+
+  public render() {
+    const { vpcOptions, subnets, securityGroups } = this.state;
+    const { values } = this.props.formik;
+    const subnetOptions = (subnets || []).map(this.toSubnetOption);
+    const sgOptions = securityGroups ? this.getSecurityGroupsByVpc(securityGroups) : [];
+    return (
+      <div className="form-group">
+        <div className="col-md-11">
+          <div className="sp-margin-m-bottom">
+            {values.credentials && (
+              <FormikFormField
+                name="vpcId"
+                label="VPC Id"
+                help={<HelpField id="aws.function.vpc.id" />}
+                fastField={false}
+                input={props => (
+                  <ReactSelectInput
+                    {...props}
+                    stringOptions={vpcOptions
+                      .filter((v: IVpc) => v.account === values.credentials)
+                      .map((v: IVpc) => v.id)}
```

Comment:
Not a task for today, but we generally don't show `vpc.id` in the UI, we try to use friendly names (like `vpc0` at netflix).  However, I don't know the details of this mapping off the top of my head.

Reply:
Not all VPCs returned by the API `getAllVps()` have a name. All have the ID field.
I agree it is not user friendly. I will see to it that this is fixed in the future.